### PR TITLE
[Spot] Make spot_state use safe_cursor().

### PR DESCRIPTION
**Pull request by concretevitamin**
When developing a basic Flask app that eventually calls into `spot_state.get_spot_jobs()`, SQLite complained about threading errors. With `safe_cursor()` this error was fixed. This PR changes the entire spot_state module to use it.

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [x] All smoke tests: `pytest tests/test_smoke.py --managed-spot` 
  - running
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/spot/spot_state.py [hunk 1]
```diff
@@ -18,9 +19,25 @@
...
 _DB_PATH.parents[0].mkdir(parents=True, exist_ok=True)
 
+# Module-level connection/cursor; thread-safe as the module is only imported
+# once.
 _CONN = sqlite3.connect(str(_DB_PATH))
 _CURSOR = _CONN.cursor()
 
+
+@contextlib.contextmanager
+def _safe_cursor():
```
Comment (Michaelvll) [inline sky/spot/spot_state.py:29]: It seems we have this in the `sky.skylet.configs` as well. Do we want to refactor it as a common utility function in `sky.utils.db_utils`?

Also, we have a `db_utils.SQLiteConn` which creates a thread-level connection for the database and serves for the same purpose in `job_lib` and `global_user_state`. Can we have choose one of them for all the modules using sqlite?

**Review by Michaelvll — COMMENTED** [review]
Thanks for identifying this issue @concretevitamin! Left a question about the consistency across the modules. : )

File: sky/spot/spot_state.py (hunk 1 above)
Comment (concretevitamin) [reply sky/spot/spot_state.py]: Good point - moved it into `db_utils.safe_cursor()`. `SQLiteConn` doesn't come with auto-commit as a context manager so I didn't use it.

**Review by Michaelvll — APPROVED** [review]
Thanks @concretevitamin! LGTM.

One general question: will adding the `commit` after read-only queries degrade the performance, as we have quite many `SELECT` queries that do no modify the database?

**Comment by concretevitamin** [comment]
> Thanks @concretevitamin! LGTM.
> 
> One general question: will adding the `commit` after read-only queries degrade the performance, as we have quite many `SELECT` queries that do no modify the database?

On a controller,
```
%timeit spot_state.get_spot_jobs()
513 µs ± 5.94 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```
which is fast enough.

**Pull request merged**